### PR TITLE
Add bastyl mini keyboard

### PR DESCRIPTION
**Pull request by bstiq**
## Description

Add the files for the Bastyl Mini keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

.

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

Regards,

### CODE REVIEW

File: keyboards/bastardkb/bastylmini/config.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#define MATRIX_COL_PINS { B4, E6, C6, B1, B3, B2 }
+#define RGB_DI_PIN D2
+#define RGBLED_NUM 21
+#define RGBLIGHT_SPLIT
+#define RGBLIGHT_ANIMATIONS
+#define DEBOUNCE 5
+#define SOFT_SERIAL_PIN D0
+#define LOCKING_SUPPORT_ENABLE
+#define LOCKING_RESYNC_ENABLE
+#define MASTER_RIGHT
```
Comment (drashna) [inline keyboards/bastardkb/bastylmini/config.h:39]: Just some readability nitpicking. 
```suggestion
#define MATRIX_ROWS 8
#define MATRIX_COLS 6

#define MATRIX_ROW_PINS { B5, F7, F6, B6 }
#define MATRIX_COL_PINS { B4, E6, C6, B1, B3, B2 }

#define DIODE_DIRECTION ROW2COL

#define RGB_DI_PIN D2
#define RGBLED_NUM 21
#define RGBLIGHT_SPLIT
#define RGBLIGHT_ANIMATIONS

#define DEBOUNCE 5

#define SOFT_SERIAL_PIN D0

#define LOCKING_SUPPORT_ENABLE
#define LOCKING_RESYNC_ENABLE

#define MASTER_RIGHT
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/bastardkb/bastylmini/config.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#define MATRIX_COLS 6
+
+#define MATRIX_ROW_PINS { B5, F7, F6, B6 }
+#define MATRIX_COL_PINS { B4, E6, C6, B1, B3, B2 }
+
+#define DIODE_DIRECTION ROW2COL
+
+#define RGB_DI_PIN D2
+#define RGBLED_NUM 21
+#define RGBLIGHT_SPLIT
```
Comment (drashna) [inline keyboards/bastardkb/bastylmini/config.h:36]: If this is per key, 

```suggestion
#define RGBLED_NUM 42
#define RGBLED_SPLIT { 21, 21 }
```

But in this case, rgb matrix would be the better way to go, though, it's not as well supported, ATM.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bastardkb/bastylmini/bastylmini.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+#pragma once
+#include "quantum.h"
+
+#define LAYOUT( \
+	k00, k01, k02, k03, k04, k05,			k45, k44, k43, k42, k41, k40, \
+	k10, k11, k12, k13, k14, k15,			k55, k54, k53, k52, k51, k50, \
+	k20, k21, k22, k23, k24, k25,			k65, k64, k63, k62, k61, k60, \
+		       k33, k34, k31,		        k71, k74, k73z \
```
Comment (noroadsleft) [inline keyboards/bastardkb/bastylmini/bastylmini.h:26]: Typo here.

```suggestion
		       k33, k34, k31,		        k71, k74, k73  \
```

File: keyboards/bastardkb/bastylmini/info.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{
+    "keyboard_name": "Bastyl Mini",
+    "url": "https://www.bastardkb.com",
+    "maintainer": "Quentin Lebastard",
+    "width": 17,
+    "height": 8,
```
Comment (noroadsleft) [inline keyboards/bastardkb/bastylmini/info.json:6]:
```suggestion
    "height": 4,
```

File: keyboards/bastardkb/bastylmini/rules.mk [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
```
Comment (noroadsleft) [inline keyboards/bastardkb/bastylmini/rules.mk:20]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/bastardkb/bastylmini/config.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+#include "config_common.h"
+#define VENDOR_ID 0xA8F8
+#define PRODUCT_ID 0x1828
+#define DEVICE_VER 0x0001
+#define MANUFACTURER Bastard Keyboards
+#define PRODUCT Bastyl Mini
+
+#define MATRIX_ROWS 8
+#define MATRIX_COLS 6
+
```
Comment (NullSense) [inline keyboards/bastardkb/bastylmini/config.h:28]:
```suggestion
#define RGBLIGHT_LIMIT_VAL 180
```
I own this keyboard and this is the max brightness I found the keyboard to run at a white color.

File: keyboards/bastardkb/bastylmini/config.h [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+#define RGBLIGHT_SPLIT
+#define RGBLIGHT_ANIMATIONS
+
+#define DEBOUNCE 5
+
+#define SOFT_SERIAL_PIN D0
+
+#define LOCKING_SUPPORT_ENABLE
+#define LOCKING_RESYNC_ENABLE
+
```
Comment (NullSense) [inline keyboards/bastardkb/bastylmini/config.h:45]:
```suggestion
#define ONESHOT_TIMEOUT 900
#define F_SCL 400000L
#define USB_POLLING_INTERVAL_MS 1
```

Some extra performance.

File: keyboards/bastardkb/bastylmini/rules.mk [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+SPLIT_KEYBOARD = yes
```
Comment (NullSense) [inline keyboards/bastardkb/bastylmini/rules.mk:23]:
```suggestion
SPLIT_KEYBOARD = yes
TAP_DANCE_ENABLE = yes
LTO_ENABLE = yes
```

Binary size reduction and tap dancing.

File: keyboards/bastardkb/bastylmini/rules.mk [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (NullSense) [inline keyboards/bastardkb/bastylmini/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes            # USB Nkey Rollover
```

**Comment by bstiq** [comment]
Hi, the CI fail seems to be related to the tap dance. I don't know anything about tap dance, should I just remove the define from the rules ? @NullSense @drashna 

Thanks for your help !

File: keyboards/bastardkb/bastylmini/config.h [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+#define RGBLED_SPLIT { 21, 21 }
+#define RGBLIGHT_ANIMATIONS
+
+#define DEBOUNCE 5
+
+#define SOFT_SERIAL_PIN D0
+
+#define LOCKING_SUPPORT_ENABLE
+#define LOCKING_RESYNC_ENABLE
+#define ONESHOT_TIMEOUT 900
```
Comment (NullSense) [inline keyboards/bastardkb/bastylmini/config.h:45]: Now that I think about it this might be too large for a default configuration.

**Comment by NullSense** [comment]
> Hi, the CI fail seems to be related to the tap dance. I don't know anything about tap dance, should I just remove the define from the rules ? @NullSense @drashna
> 
> Thanks for your help !

I'm not sure why this is happening, maybe the others can help.

**Comment by NullSense** [comment]
https://github.com/NullSense/dots/blob/master/readme.md you could also add a nice readme for users new to the keyboard to easily preview the layout (or adding comments above each layer).

File: keyboards/bastardkb/bastylmini/rules.mk [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+SPLIT_KEYBOARD = yes
+TAP_DANCE_ENABLE = yes
```
Comment (NullSense) [inline keyboards/bastardkb/bastylmini/rules.mk:24]: Turns out this feature doesn't work if there's no tap dance definitions, so bettter omit this line, that's why CI fails @bstiq.

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**